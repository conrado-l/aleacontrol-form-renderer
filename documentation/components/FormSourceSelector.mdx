---
name: FormSourceSelector
---

import { Playground, PropsTable } from 'docz'
import FormSourceSelector from '../../src/components/ActionButton.js'

# FormSourceSelector

<PropsTable of={FormSourceSelector} />

## Description
A button for changing the form data source samples
## Demo and usage

<Playground>
    <FormSourceSelector changeFormSample={(e) => e}/>
</Playground>
