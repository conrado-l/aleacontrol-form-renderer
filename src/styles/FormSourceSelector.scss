@import "variables";
@import "mixins";

.form-source-selector {
  height: 60px;
  width: 300px;
  padding: 5px;
  background-color: $box-background-color;
  border-radius: $box-border-radius;
  @include box-shadow;
}

.change-data-button {
  height: 50px;
}
