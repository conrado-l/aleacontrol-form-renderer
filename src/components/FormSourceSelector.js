import React, { Component } from 'react'
import Button from 'react-bootstrap/Button'
import '../styles/FormSourceSelector.scss'
import PropTypes from 'prop-types'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'

/**
 * Button that should change the current form
 */
class FormSourceSelector extends Component {
  render () {
    return (
      <Row>
        <Col xs={12} md={8} xl={6}>
          <div className='form-source-selector w-100 mt-2 d-flex align-content-center justify-content-around'>
            <Button variant='outline-info' className='change-data-button' onClick={this.props.changeFormSample}>
                Change Form Sample</Button>
          </div>
        </Col>
      </Row>
    )
  }
}

FormSourceSelector.propTypes = {
  changeFormSample: PropTypes.func
}
export default FormSourceSelector
