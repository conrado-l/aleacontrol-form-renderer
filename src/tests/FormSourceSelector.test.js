import React from 'react'
import { mount } from 'enzyme'
import FormSourceSelector from '../components/FormSourceSelector'
import Button from 'react-bootstrap/Button'
import renderer from 'react-test-renderer'

describe('FormSourceSelector', () => {
  it('should render correctly form source selector component', () => {
    const clickFn = jest.fn()
    const FormSourceSelectorComponent = renderer
      .create(<FormSourceSelector changeFormSample={clickFn} />)
      .toJSON()
    expect(FormSourceSelectorComponent).toMatchSnapshot()
  })

  it('should render the button', () => {
    const FormSourceSelectorComponent = mount(<FormSourceSelector />)

    expect(FormSourceSelectorComponent.find(Button)).toHaveLength(1)
  })

  it('should call the callback function on click', () => {
    const clickFn = jest.fn()
    const FormSourceSelectorComponent = mount(<FormSourceSelector changeFormSample={clickFn} />)

    FormSourceSelectorComponent.find(Button).simulate('click')

    expect(clickFn).toHaveBeenCalled()
  })
})
