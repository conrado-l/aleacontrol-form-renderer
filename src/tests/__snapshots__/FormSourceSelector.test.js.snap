// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSourceSelector should render correctly form source selector component 1`] = `
<div
  className="row"
>
  <div
    className="col-xl-6 col-md-8 col-12"
  >
    <div
      className="form-source-selector w-100 mt-2 d-flex align-content-center justify-content-around"
    >
      <button
        className="change-data-button btn btn-outline-info"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        Change Form Sample
      </button>
    </div>
  </div>
</div>
`;
